### PR TITLE
CMP-2543: Implement whitelist for configure-network-policies-namespaces

### PR DESCRIPTION
**Pull request by Vincent056**
Added a new varible var_network_policies_namespaces_whitelist_regex, and updated the rule `configure_network_policies_namespaces` so user is able to excude namespaces by setting this variable to the regex of namespace they want to exclude for this rule

### CODE REVIEW

File: applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var [hunk 1]
```diff
@@ -0,0 +1,18 @@
+documentation_complete: true
+
+title: 'Whitelist Namespaces for Network Policies'
```
Comment (yuumasato) [inline applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var:3]: Let's use more inclusive and descriptive wording:

```suggestion
title: 'Namespaces Allowed Without Network Policy'
```

or
```suggestion
title: 'Namespaces exempt of Network Policies'
```

Don't  forget to rename the variable file accordingly.

File: applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Namespaces regular expression explicitly allowed
+    through network policy filters, e.g. setting value to
+    "namespace1|namespace2" will whitelist namespace
+    "namespace1" and "namespace2" for network policies checks.
+
+type: string
+
+operator: equals
+
+interactive: false
```
Comment (yuumasato) [inline applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var:15]: I guess this field is not consumed by CO, but if this should be editable by the user, this should be `true`
```suggestion
interactive: true
```

**Review by yuumasato — CHANGES_REQUESTED** [review]
Please, mention in the rule description as well that namespaces matching the variable regex are not checked for presence of a Network Policy.

File: applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var (hunk 1 above)
Comment (yuumasato) [reply applications/openshift/networking/var_network_policies_namespaces_whitelist_regex.var]: @Vincent056 You forgot to rename the variable and the references to it, :grin:

**Review by yuumasato — CHANGES_REQUESTED** [review]
Please mention the variable in the rule description.

I have tried the rule in the cluster and it is not working for me, :(
It is always incompliant.

**Comment by Vincent056** [comment]
@yuumasato thanks for the review, this issue should be fixed now with the new commit.

```
$ oc create ns no-policy-0
$ oc create ns no-policy-1

$ ./utils/build_ds_container.py -p -d -P ocp4


cat <<EOF | oc apply -f -
apiVersion: compliance.openshift.io/v1alpha1
kind: TailoredProfile
metadata:
  name: cis-pass
  namespace: openshift-compliance
spec:
  description: Test
  setValues:
  - name: upstream-ocp4-var-network-policies-namespaces-whitelist-regex
    value: no-policy-0|no-policy-1
    rationale: test
  extends: upstream-ocp4-cis
  title: My modified nist profile with a custom value
EOF

cat <<EOF | oc apply -f -
apiVersion: compliance.openshift.io/v1alpha1
kind: TailoredProfile
metadata:
  name: cis-no-pass
  namespace: openshift-compliance
spec:
  description: Test
  extends: upstream-ocp4-cis
  title: My modified nist profile with a custom value
EOF

[vincent@node cac-content-fork]$ oc get tp
NAME          STATE
cis-no-pass   READY
cis-pass      READY

cat <<EOF | oc apply -f -
apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: cis-test-network-policy
  namespace: openshift-compliance
profiles:
- apiGroup: compliance.openshift.io/v1alpha1
  kind: TailoredProfile
  name: cis-pass
- apiGroup: compliance.openshift.io/v1alpha1
  kind: TailoredProfile
  name: cis-no-pass
settingsRef:
  apiGroup: compliance.openshift.io/v1alpha1
  kind: ScanSetting
  name: default
EOF

[vincent@node cac-content-fork]$ oc get ssb
NAME                      STATUS
cis-test-network-policy   READY

[vincent@node cac-content-fork]$ oc get scan -w
NAME          PHASE     RESULT
cis-no-pass   RUNNING   NOT-AVAILABLE
cis-pass      RUNNING   NOT-AVAILABLE
cis-pass      AGGREGATING   NOT-AVAILABLE
cis-pass      AGGREGATING   NOT-AVAILABLE
cis-no-pass   AGGREGATING   NOT-AVAILABLE
cis-no-pass   AGGREGATING   NOT-AVAILABLE
cis-no-pass   DONE          NON-COMPLIANT
cis-pass      DONE          NON-COMPLIANT

[vincent@node cac-content-fork]$ oc get ccr | grep network-policies-n
cis-no-pass-configure-network-policies-namespaces                    FAIL     high
cis-pass-configure-network-policies-namespaces                       PASS     high

oc delete ssb cis-test-network-policy
oc delete tp cis-no-pass cis-pass
oc delete profilebundle upstream-ocp4
```

**Review by yuumasato — APPROVED** [review]

**Pull request merged**